src/test/java/com/hillel/items_exchange/security/SecurityConfigIntegrationTest.java

```diff
@@ -89,34 +102,84 @@ public void loginWithNotValidUserIsOk() throws Exception {
         mockMvc.perform(post("/adv"))
                 .andExpect(status().isUnauthorized());
     }
 
     @Test
+    @Transactional
     @DataSet("database_init.yml")
     public void createAdvertisementWithValidTokenWithoutAdvertisementDtoIsBadRequest() throws Exception {
-        final String token = obtainToken(validLoginDto);
+        final String validToken = "Bearer " + obtainToken(validLoginDto);
+
         mockMvc.perform(post("/adv")
-                .header("Authorization", "Bearer " + token))
+                .header("Authorization", validToken))
                 .andExpect(status().isBadRequest());
     }
 
     @Test
     @Transactional
     @DataSet("database_init.yml")
-    @ExpectedDataSet(value = "advertisement/create.yml", ignoreCols = {"created", "updated"})
+    @ExpectedDataSet(value = "security/create-advertisement.yml", ignoreCols = {"created", "updated"})
     public void createAdvertisementWithValidTokenAndValidAdvertisementDtoIsOk() throws Exception {
-        final String token = obtainToken(validLoginDto);
+        final String validToken = "Bearer " + obtainToken(validLoginDto);
+
         mockMvc.perform(post("/adv")
-                .header("Authorization", "Bearer " + token)
+                .header("Authorization", validToken)
                 .content(asJsonString(nonExistDto))
                 .contentType(MediaType.APPLICATION_JSON)
                 .accept(MediaType.APPLICATION_JSON))
                 .andExpect(status().isCreated());
     }
 
+    @Test
+    @DataSet("database_init.yml")
+    public void postRequestWithJWTTokenWithoutBearerPrefixIsUnauthorizedAndBearerIsAbsent() throws Exception {
+        final String tokenWithoutBearerPrefix = obtainToken(validLoginDto);
+
+        String errorMessage = mockMvc.perform(post("/adv")
+                .header("Authorization", tokenWithoutBearerPrefix))
+                .andExpect(status().isUnauthorized())
+                .andReturn()
+                .getResponse()
+                .getErrorMessage();
+
+        assertEquals(jwtTokenNotStartWithBearer, errorMessage);
+    }
+
+    @Test
+    @DataSet("database_init.yml")
+    public void postRequestWithNotValidJWTTokenIsUnauthorizedAndBadTokenSignature() throws Exception {
+        final String invalidToken = "Bearer " + obtainToken(validLoginDto).replaceAll(".$", "");
+
+        String errorMessage = mockMvc.perform(post("/adv")
+                .header("Authorization", invalidToken))
+                .andExpect(status().isUnauthorized())
+                .andReturn()
+                .getResponse()
+                .getErrorMessage();
+
+        assertEquals(jwtTokenSignatureNotValid, errorMessage);
+    }
+
+    @Test
+    @DataSet("database_init.yml")
+    public void postRequestWithExpiredJwtTokenIsUnauthorizedAndTokenIsExpired() throws Exception {
+        final String validToken = "Bearer " + obtainToken(validLoginDto);
+        TimeUnit.MILLISECONDS.sleep(jwtTimeExpired);
```

Comment:
You shouldn't sleep the main thread. Is there no other way to do it?

Reply:
Unfortunately, I have not found another way to check token validity. To optimize this path I'm set token expired time to 2 seconds.